lib/core/versions/latest/FeeManager.ts

```diff
@@ -0,0 +1,57 @@
+import ErrorCode from './ErrorCode';
+import ProtocolParameters from './ProtocolParameters';
+import { SidetreeError } from '../../Error';
+
+/**
+ * Encapsulates the functionality to calculate and verify the blockchain transaction fees.
+ */
+export default class FeeManager {
+
+  /**
+   * Converts the normalized fee (returned by the blockchain) into the transaction fee to be paid when writing
+   * the current transaction.
+   *
+   * @param normalizedFee The normalized fee for the current transaction.
+   * @param numberOfOperations The number of operations to write.
+   * @param feeMarkupFactor Markup to be added to the calculated fee.
+   *
+   * @throws if the number of operations are <= 0.
+   */
+  public static convertNormalizedFeeToTransactionFee (normalizedFee: number, numberOfOperations: number, feeMarkupFactor: number): number {
+
+    if (numberOfOperations <= 0) {
+      throw new SidetreeError(ErrorCode.OperationCountLessThanZero, `Fee cannot be calculated for the given number of operations: ${numberOfOperations}`);
+    }
+
+    const normalizedFeePerOperation = normalizedFee * ProtocolParameters.normalizedToPerOperationFeeFactor;
+    const perOperationFee = normalizedFeePerOperation * numberOfOperations;

```

Comment:
`perOperationFee` sounds like it is holding the "fee per operation", which is obviously not true.
```suggestion
    const minimumTransactionFee= normalizedFeePerOperation * numberOfOperations;
```